crates/app/src/handle.rs

```diff
@@ -15,7 +15,18 @@ use tower::Service;
 /// Parses the URI of `req` and routes it to respective component.
 pub async fn handle(mut req: Request<Body>) -> impl IntoResponse {
     trace!(target: "app::handle", "begin HTTP request handling {:?}", req);
-    let path = req.uri().path().strip_prefix('/').expect("invalid URI");
+    let path = req.uri().path().trim_start_matches('/');
+    let path = path
+        .strip_prefix("api")
+        .ok_or((StatusCode::NOT_FOUND, format!("Route `/{path}` not found")))?
+        .trim_start_matches('/')
+        // TODO: Parse SemVer, support v0, v0.1 etc.
+        .strip_prefix("v0.1.0")
```

Comment:
Normally in API versions I just see `v1`, `v2`, `v3`, no minor or patch numbers. Do you think that it will be possible to meaningfully version this API while making use of minor and patch versions?

Reply:
What's the benefit here with this level of granularity?